flink-clients/src/main/java/org/apache/flink/client/ClientUtils.java

```diff
@@ -85,7 +86,7 @@ public static void executeProgram(
             PackagedProgram program,
             boolean enforceSingleJobExecution,
             boolean suppressSysout)
-            throws ProgramInvocationException {
+            throws ProgramInvocationException, JobStartupFailedException {
```

Comment:
These exception sound like the same to me. Maybe `JobStartupFailedException extends ProgramInvocationException`?
Now thinking about this, isn't `JobStartupFailedException` implying that the job's startup didn't work (for example cluster is too small)? Wouldn't a better name be `Job/ProgramDefinitionException`?

Reply:
Can you take a look at 
feefcb7? Unfortunately, I had to move the `ProgramInvocationException` that the `JobStartupFailedException` can extend it because there is for a good reason no `clients` -> `runtime` dependency.

I am a bit worried that theoretically `ProgramInvocationException` was user-facing but not annotated.

Reply:
I also renamed the exception to `FatalProgramInvocationException` because it does not cause a possible restart if HA is configured.